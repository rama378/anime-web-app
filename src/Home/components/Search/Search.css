.Search {
    position: relative;
    padding: 32px 0 0 0;
    background-color: var(--background-dark);
}

.Search__wrapper {
    height: 150px;
    width: 60%;
    margin: 0 auto;
    background: transparent url('../../images/search-image.png') no-repeat;
    background-position: 90% -45px;
    background-size: 300px;
}

.Search__title {
    width: 50%;
    font-size: var(--font-size-extralarge);
    font-weight: bold;
}

.Search__inputwrapper {
    position: absolute;
    bottom: -30px;
    width: calc(100% - 36px);
    text-align: center;
}

.Search__input {
    width: 60%;
    margin: 0 auto;
}

@media only screen and (max-width: 500px) {
    .Search__title {
        width: 100%;
    }

    .Search__wrapper {
        width: calc(100% - 64px);
        background: none;
    }

    .Search__input {
        width: calc(100% - 64px);
    }
}