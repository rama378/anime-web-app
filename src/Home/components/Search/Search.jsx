import Input from '../../../App/common/components/Input/Input';
import './Search.css';
import {ReactComponent as SearchIcon} from "../../../App/common/assets/images/icon/search.svg";

const Search = () => {
  return (
    <div className='Search'>
        <div className='Search__wrapper'>
            <div className='Search__title'>Find your anime</div>
            <div>e.g. Cowboy Bebop, Trigun, Witch Hunter Robin</div>
        </div>
        <div className='Search__inputwrapper'>
            <Input
                size="large"
                className='Search__input'
                placeholder='Type your anime title...'
                icon={<SearchIcon />}
            />
        </div>
    </div>
  );
}

export default Search;
