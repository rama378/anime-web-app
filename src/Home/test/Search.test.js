import React from 'react';
import ShallowRenderer from 'react-test-renderer/shallow';

import Search from '../components/Search/Search';

const renderer = new ShallowRenderer();

describe('<Search />', () => {
  it('should render and match the snapshot no prop given', () => {
    renderer.render(<Search />);
    const renderedOutput = renderer.getRenderOutput();
    expect(renderedOutput).toMatchSnapshot();
  });
});
