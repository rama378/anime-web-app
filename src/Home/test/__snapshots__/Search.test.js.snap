// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Search /> should render and match the snapshot no prop given 1`] = `
<div
  className="Search"
>
  <div
    className="Search__wrapper"
  >
    <div
      className="Search__title"
    >
      Find your anime
    </div>
    <div>
      e.g. Cowboy Bebop, Trigun, Witch Hunter Robin
    </div>
  </div>
  <div
    className="Search__inputwrapper"
  >
    <Input
      className="Search__input"
      icon={<ForwardRef(SvgSearch) />}
      placeholder="Type your anime title..."
      size="large"
    />
  </div>
</div>
`;
